src/lib/autodiscovery/autodiscovery_manager.py

```diff
@@ -1,76 +1,148 @@
-import asyncio
-from datetime import datetime, timedelta
-from typing import Any, Dict, List
+from typing import Any, Dict, List, Optional
+from dataclasses import dataclass
 
-from lib.autodiscovery.autodiscovery import enrich_services_with_autodiscovery_metrics
-from lib.configuration import config
-from lib.metrics import GCPService
+import yaml
 
-AUTODISCOVERY_QUERY_INTERVAL_SEC = config.get_autodiscovery_querry_interval() * 60
+from lib.autodiscovery.autodiscovery import (
+    AutodiscoveryResourceLinking,
+    enrich_services_with_autodiscovery_metrics,
+)
+from lib.context import LoggingContext
+from lib.metrics import AutodiscoveryGCPService, GCPService
+from lib.utilities import read_autodiscovery_block_list_yaml, read_autodiscovery_config_yaml
+from pathlib import Path
+
+
+@dataclass(frozen=True)
+class ServiceStub:
+    extension_name: str
+    service_name: str
+    feature_set_name: str
 
 
 class AutodiscoveryManager:
-    autodiscovery_task: asyncio.Task
-    autodiscovered_metrics_cache: List[GCPService]
-    autodiscovered_extension_versions_hash: int
-    time_since_last_autodiscovery: datetime
-    query_interval = timedelta(seconds=AUTODISCOVERY_QUERY_INTERVAL_SEC)
+    autodiscovery_config: Dict[str, Any]
+    autodiscovery_resource_mapping: Dict[str, List[ServiceStub]]
+    autodiscovery_metric_block_list: List[str]
     last_autodiscovered_metric_list_names: Dict[str, Any]
+    logging_context = LoggingContext("AUTODISCOVERY_MANAGER")
+    autodiscovery_enabled: bool
 
-    @staticmethod
-    async def init(services: List[GCPService], current_extension_versions: Dict[str, str]):
-        autodiscovery_manager = AutodiscoveryManager(services, current_extension_versions)
-        await autodiscovery_manager._refresh_autodiscovery_task(
-            services, current_extension_versions
-        )
-        return autodiscovery_manager
-
-    def __init__(self, services: List[GCPService], current_extension_versions: Dict[str, str]):
+    def __init__(self):
+        self.autodiscovery_config = {}
+        self.autodiscovery_resource_mapping = {}
+        self.autodiscovery_metric_block_list = []
         self.last_autodiscovered_metric_list_names = {}
-        self.autodiscovered_metrics_cache = []
-        self.autodiscovered_extension_versions_hash = hash(
-            tuple(sorted(current_extension_versions.items()))
-        )
-        self.query_interval = timedelta(seconds=AUTODISCOVERY_QUERY_INTERVAL_SEC)
-        self.autodiscovery_task = asyncio.create_task(
-            enrich_services_with_autodiscovery_metrics(
-                services, self.last_autodiscovered_metric_list_names
+        self.autodiscovery_enabled = True
+
+        self._load_config()
+
+    def _load_config(self):
+        try:
+            self.autodiscovery_config = read_autodiscovery_config_yaml()["autodicovery_config"]
+
+            with open("./lib/autodiscovery/config/autodiscovery-mapping.yaml", "r") as file_mapping:
```

Comment:
How will the maintenance of the autodiscovery-mapping look like? What if we change our extensions, or add more? Will we have to change this too?